### ISSUE TITLE
simulation error while giving velocity profile at inlet

### ISSUE THREAD

**Issue by Shail546** [turbinesFoam/turbinesFoam#341]
Sir,
I am trying to give a velocity profile at inlet (say parabolic velocity profile) using codedFixedValue technique for inlet in U file. But it shows "floating point error".
I tried the same with the tutorials provided with openFoam (say pitzDaily) and this works fine with them.
Could you please help me with this issue. I am attaching the code and the error file for your reference.
Thank you.
![Screenshot from 2021-12-26 20-21-31](https://user-images.githubusercontent.com/87179388/147411838-c48f64a0-ba39-4ed6-8c20-7bc689a5df67.png)
![Screenshot from 2021-12-20 15-42-30](https://user-images.githubusercontent.com/87179388/147411845-668928de-c5e5-49b6-a361-c0805d4267d3.png)

**Comment by petebachant** [comment]
Have you tried running this same case with `fvOptions` disabled?

**Comment by Shail546** [comment]
Sir, I need fvOptions to add more turbines in the domain. So I can't disable it. But it is showing same error for single turbine as well as for multiple turbines.

**Comment by Shail546** [comment]
> Have you tried running this same case with `fvOptions` disabled?

Sir I tried with this also but again I am getting the same error.

**Comment by petebachant** [comment]
> Sir I tried with this also but again I am getting the same error.

Alright, so it sounds like an issue with the case itself but not the actuator lines.

**Comment by Shail546** [comment]
> > Sir I tried with this also but again I am getting the same error.
> 
> Alright, so it sounds like an issue with the case itself but not the actuator lines.

Sir please provide me some possible solution of the problem. Thank you

**Comment by Shail546** [comment]
> Have you tried running this same case with `fvOptions` disabled?

Sir

**Comment by Shail546** [comment]
> > Sir I tried with this also but again I am getting the same error.
> 
> Alright, so it sounds like an issue with the case itself but not the actuator lines.

Sir, It was the lower limit of z- axis in blockMesh that was creating the problem. It is not taking negative range, So I chnged the lower limit of -1.22 to 0, But now fvOption is creating the problem. Now I ran the simulation by disabling the fvOption file and the simulation ran perfectly but with fvOption it is showing the same error as earlier.

**Comment by petebachant** [comment]
Can you upload your case to a GitHub repo and share here?

**Comment by Shail546** [comment]
> Can you upload your case to a GitHub repo and share here?

https://github.com/Shail546/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/tree/main/codedFixedValue

Sir, I have uploaded my case in this depository. Please check it. 
It is the case of twin rotor VAWT using SST k-Omega model. In this I am giving a particular velocity profile at inlet using codedFixedValue. But fvOption is creating the problem.
Thank you.

**Comment by petebachant** [comment]
I tried running your case and it looks like your boundary condition code isn't able to compile properly (below is from `log.pimpleFoam`:

```c++
Using dynamicCode for patch inlet on field U at line 31 in "/home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/0/U.boundaryField.inlet"
Could not load "/home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so"
/home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so: cannot open shared object file: No such file or directory
Creating new library in "dynamicCode/velocityProfile/platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so"
Invoking wmake libso /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/velocityProfile
wmake libso /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/velocityProfile
    ln: ./lnInclude
    dep: fixedValueFvPatchFieldTemplate.C
    Ctoo: fixedValueFvPatchFieldTemplate.C
fixedValueFvPatchFieldTemplate.C: In copy constructor ‘Foam::velocityProfileFixedValueFvPatchVectorField::velocityProfileFixedValueFvPatchVectorField(const Foam::velocityProfileFixedValueFvPatchVectorField&)’:
fixedValueFvPatchFieldTemplate.C:135:1: warning: base class ‘class Foam::dictionaryContent’ should be explicitly initialized in the copy constructor [-Wextra]
  135 | velocityProfileFixedValueFvPatchVectorField::
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ld: /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/velocityProfile/../platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so
```

**Comment by Shail546** [comment]
> I tried running your case and it looks like your boundary condition code isn't able to compile properly (below is from `log.pimpleFoam`:
> 
> ```c++
> Using dynamicCode for patch inlet on field U at line 31 in "/home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/0/U.boundaryField.inlet"
> Could not load "/home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so"
> /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so: cannot open shared object file: No such file or directory
> Creating new library in "dynamicCode/velocityProfile/platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so"
> Invoking wmake libso /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/velocityProfile
> wmake libso /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/velocityProfile
>     ln: ./lnInclude
>     dep: fixedValueFvPatchFieldTemplate.C
>     Ctoo: fixedValueFvPatchFieldTemplate.C
> fixedValueFvPatchFieldTemplate.C: In copy constructor ‘Foam::velocityProfileFixedValueFvPatchVectorField::velocityProfileFixedValueFvPatchVectorField(const Foam::velocityProfileFixedValueFvPatchVectorField&)’:
> fixedValueFvPatchFieldTemplate.C:135:1: warning: base class ‘class Foam::dictionaryContent’ should be explicitly initialized in the copy constructor [-Wextra]
>   135 | velocityProfileFixedValueFvPatchVectorField::
>       | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     ld: /home/pete/OpenFOAM/pete-v2106/run/Inlet-velocity-profile-for-twin-rotor-VAWT-using-SST-k-Omega-model/codedFixedValue/dynamicCode/velocityProfile/../platforms/linux64GccDPInt32Opt/lib/libvelocityProfile_bdae92d9bc1276b1fbf1e37265706700cc617707.so
> ```

yes sir I am also getting the same error. But I am not able to solve the issue.

**Comment by lpz456** [comment]
Hi, friends, Have you solve your problem? If so, can you share your experience?
